packages/ui-color-picker/src/ColorPreset/index.tsx

```diff
@@ -0,0 +1,299 @@
+
+/** @jsx jsx */
+import { Component, SyntheticEvent } from 'react'
+
+import { passthroughProps } from '@instructure/ui-react-utils'
+import { withStyle, jsx } from '@instructure/emotion'
+import { IconButton, Button } from '@instructure/ui-buttons'
+import { View } from '@instructure/ui-view'
+import { Tooltip } from '@instructure/ui-tooltip'
+import { Popover } from '@instructure/ui-popover'
+import { Text } from '@instructure/ui-text'
+import { Drilldown } from '@instructure/ui-drilldown'
+import type { DrillDownOnSelectArgs } from '@instructure/ui-drilldown'
+import { IconAddLine, IconCheckDarkSolid } from '@instructure/ui-icons'
+import { colorToHex8, hexToRgb } from '@instructure/ui-color-utils'
+import { ColorIndicator } from '../ColorIndicator'
+
+import type { ColorPresetProps, ColorPresetState } from './props'
+import generateStyle from './styles'
+import generateComponentTheme from './theme'
+import ColorMixer from '../ColorMixer'
+import ColorContrast from '../ColorContrast'
+/**
+---
+category: components
+---
+@tsProps
+**/
+@withStyle(generateStyle, generateComponentTheme)
+class ColorPreset extends Component<ColorPresetProps, ColorPresetState> {
+  constructor(props: ColorPresetProps) {
+    super(props)
+    this.state = {
+      openEditor: false,
+      openAddNew: false,
+      newColor: { r: 51, g: 99, b: 42, a: 1 }
+    }
+  }
+  static defaultProps = {
+    disabled: false
+  }
+  ref: HTMLDivElement | null = null
+
+  handleRef = (el: HTMLDivElement | null) => {
+    const { elementRef } = this.props
+
+    this.ref = el
+
+    if (typeof elementRef === 'function') {
+      elementRef(el)
+    }
+  }
+
+  componentDidMount() {
+    this.props.makeStyles?.()
+  }
+
+  componentDidUpdate() {
+    this.props.makeStyles?.()
+  }
+
+  onMenuItemSelected =
+    (color: string) =>
+    (_e: SyntheticEvent<Element, Event>, args: DrillDownOnSelectArgs) => {
+      if (args.value === 'select') {
+        this.props.onSelect(color)
+      }
+      if (args.value === 'remove') {
+        this.props?.colorMixerSettings?.onPresetChange(
+          this.props.colors.filter((clr) => clr !== color)
+        )
+      }
+    }
+
+  renderAddNewPresetButton = () => (
+    <Popover
+      renderTrigger={
+        <div css={this.props?.styles?.addNewPresetButton}>
+          <IconButton
+            disabled={this.props.disabled}
+            screenReaderLabel={this.props.addNewPresetButtonScreenReaderLabel}
+          >
+            <IconAddLine />
+          </IconButton>
+        </div>
+      }
+      isShowingContent={this.state.openAddNew}
+      onShowContent={() => {
+        if (this.props.disabled) return
+        this.setState({ openAddNew: true })
+      }}
+      onHideContent={() => {
+        this.setState({ openAddNew: false })
+      }}
+      on="click"
+      screenReaderLabel={this.props.popoverScreenReaderLabel}
+      shouldContainFocus
+      shouldReturnFocus
+      shouldCloseOnDocumentClick
+      offsetY="16px"
```

Comment:
Use rem please. But I think it accepts number as well. Oh and please put a TODO at Position's `offsetToPx` util to fix the offset handling when it is a string (em/rem/px)